### DISCUSSION TITLE
pip install pyswmm error (how to fix this please?)

### DISCUSSION THREAD

**Discussion by BilalKemal** [pyswmm/pyswmm#398] [Q&A] [answered]
C:\WINDOWS\system32>pip install pyswmm
Collecting pyswmm
  Using cached pyswmm-1.2.0-py3-none-any.whl (101 kB)
Collecting swmm-toolkit>=0.9.0
  Using cached swmm-toolkit-0.9.0.zip (640 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\BiHa\AppData\Local\Temp\pip-install-w27t8127\swmm-toolkit_e1e158353ac743908c59aa37be3e7d47\setup.py", line 20, in <module>
          from skbuild import setup
      ModuleNotFoundError: No module named 'skbuild'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**Comment by abhiramm7** [comment]
Hi, can you confirm if you are not using 32 bit python? We don't provide the wheels for it.

**Comment by BilalKemal** [comment]
Yes I am using 64

**Comment by karosc** [comment, marked as answer]
@BilalKemal, which version of python are you using? The latest release no longer supports 3.6. So if you are using python 3.6, pip will try to build pyswmm from source, which requires a c compiler, cmake, and scikit-build to be preinstalled. 

If you can't update python beyond 3.6 and don't need to model water quality (the major addition in the latest release), you can install version 1.1.1 with `pip install pyswmm==1.1.1` 

We are actively working on a minor release that will add support for python 3.11, but for now the current pyswmm version (1.2.0) only supports python 3.7 - 3.10

**Comment by BilalKemal** [comment]
Thanks for your reply
I solved
